vortex-layout/src/scan/mod.rs

```diff
@@ -258,6 +266,11 @@ impl<A: 'static + Send> ScanBuilder<A> {
                     if mask.all_false() {
                         Ok(None)
                     } else {
+                        // If there's a limit, we only apply it at the end
+                        let mut mask = mask;
+                        if let Some(limit) = self.limit {
```

Comment:
This applies the same limit to each row split?

Reply:
that's true 🤦

Reply:
If there isn't a filter expression, then the masks should be limited at the very beginning of the scan builder (this would automatically prune splits that appeared after the limit)

If there _is_ a filter expression, then this is a reasonable approach! Apply the limit to each split's initial row mask independently. This would create an upper bound. But you also then need to add a StreamExt::scan to the end of the array stream to limit the result.

Reply:
I was really hoping to avoid `StreamExt::scan` 😂